units/host_status.go

```diff
@@ -69,16 +74,23 @@ func (j *cloudHostReadyJob) Run(ctx context.Context) {
 	if len(hostsToCheck) == 0 {
 		return
 	}
-	providers := map[string][]host.Host{}
+
+	// Collect hosts by provider and region
+	providers := map[providerInfo][]host.Host{}
 	for _, h := range hostsToCheck {
-		providers[h.Provider] = append(providers[h.Provider], h)
+		key := providerInfo{
+			provider: h.Provider,
+			region:   cloud.GetEC2Region(h.Distro.ProviderSettings),
+		}
+		providers[key] = append(providers[key], h)
 	}
```

Comment:
does it make sense to move these data conversion and grouping functions into the host package, both for better reuse, and for potentially pushing them down into aggregation operations in the future.

Reply:
Because of dependencies, I made this kind of grouping package in the cloud package (which is where ManagerOpts, the key used in the map, also now lives).